### PR TITLE
Run jest tests via Turbo

### PR DESCRIPTION
**Pull request by jamiely**
## ℹ️ What is the context for these changes?
* Uses turbo to run `test` scrip in each package (jest)
* Moves jest config and uses shared jest config in each package
* Makes `test` pipeline dependent on `build`
* Adds `test:watch` script

### Known issue with monorepo dependencies

There is a known issue (might've existed already) that may yield an unexpected test behavior.
* If there are two packages in the monorepo `A` and `B`
* and you are in a test (watching) in the context of `A`
* and you modify package `B` and affect the behavior of a test in `A`
* the test won't pickup the change b/c a build needs to be done

I tested various ways to get around this issue.

#### Solution 1

Watch files and build concurrently with the test watch.

Do this by adding a `build:watch` task defined like

```json
"build:watch": "nodemon --watch packages -e ts,tsx,json,js --exec 'yarn build' --ignore '*/dist/*'",
"test:watch": "concurrently 'yarn build:watch' 'sleep 1 && turbo run test:watch'",
```

This doesn't actually work and the test is not retriggered.

#### Solution 2

Restart the test:watch if we notice any build files changing, which will cause a new build before the test runs.

```json
"test:watch": "nodemon --exec 'turbo run test:watch' --watch packages -e ts,tsx,json,js --ignore '*/dist/*' --ignore '*.test.*'",
```

This works but is really slow when modifying files (fine when modifying tests).

#### Solution 3 (the chosen solution)

Leave as-is. Acknowledge this known issue since it is a bit of an edge case. 

Why is it an edge case? Ideally changing the behavior of a dependency should rarely affect the unit tests of the current package. When this does happen, it should be deliberate, and a build will probably be done anyway. If that didn't happen, CI should catch the problem.

## ✅ Checklist

- [ ] Tested on mobile
- [ ] Tested on multiple browsers
- [ ] Tested for accessibility
- [ ] Includes unit tests
- [x] Updated relevant documentation for the changes (if necessary)

### CODE REVIEW

**Review by caropinzonsilva — APPROVED** [review]
I agree that it is an edge case and it would be worse affecting the performance of runing other tests because of this edge case. 

I 🎩 locally and it works as expected!

File: packages/tokengate/jest.config.js [hunk 1]
```diff
@@ -0,0 +1 @@
+export {default} from '../jest/jest.config.js';
```
[the remark this answers is not in the document]
Comment (jamiely) [reply packages/tokengate/jest.config.js:1]: Hmmm... I thought that's what I was doing... maybe it would be good to pair a bit on this?

[the remark this answers is not in the document]
Comment (QuintonC) [reply packages/tokengate/jest.config.js:1]: Yep, we can 🍐 on this whenever you've got some time! :)

[the remark this answers is not in the document]
Comment (jamiely) [reply packages/tokengate/jest.config.js]: @quintonc I updated the PR with our changes if you want to take a look again.

**Review by QuintonC — APPROVED** [review]
Looks good to me. The only thing I think we _might_ be able to change in the future is adding paths to the package.json for the shared jest config package so we can make the imports shorter (e.g. `import config from 'jestconfig/base';`

That's mostly just nit/tidiness. I think this is totally fine as-is :)
